tests/internal/migrations/test_replicate_workspaces.py

```
@@ -120,31 +121,36 @@ def _do_replicate(self, stream: WorkspaceEventStream, **kwargs) -> list[Workspac
+    def _assert_event_ids(self, events: list[WorkspaceEvent], ids: Iterable[str], create_only_ids: Iterable[str] = ()):
         ids = set(ids)
+        create_only_ids = set(create_only_ids)
+
+        self.assertTrue(create_only_ids.issubset(ids))
 
         self.assertCountEqual(
             [
-                (type, id)
-                for id in ids
-                for type in [ReplicationEventType.CREATE_WORKSPACE, ReplicationEventType.UPDATE_WORKSPACE]
+                *((ReplicationEventType.CREATE_WORKSPACE, id) for id in ids),
```

Ruff (0.16.3)

[error] Variable `id` is shadowing a Python builtin

(A001)